main.py

```diff
@@ -0,0 +1,318 @@
+    )
+    census_data["pct_unemployed"] = (
+            census_data["tot_unemployed"] / census_data["B23025_001E"]
+    )
+    census_data["pct_wfh"] = census_data["tot_wfh"] / census_data["B08101_001E"]
+    census_data["pct_unenrolled"] = (
+            census_data["tot_unenrolled_school"] / census_data["tot_age_under_18"]
+    )
+    census_data["pct_wo_broadband"] = (
+            census_data["tot_wo_broadband"] / census_data["B28002_001E"]
+    )
+    census_data["pct_less_vehicles"] = census_data["tot_less_vehicle"] / (
+            census_data["B08203_001E"] - census_data["B08203_007E"]
+    )
+    census_data["pct_lang_home"] = (
+            census_data["tot_home_lang"] / census_data["C16002_001E"]
+    )
+    census_data["pct_mortgage_over35"] = (
+            census_data["tot_mortgage_over35"] / census_data["B25091_001E"]
+    )
+    census_data["pct_rent_over35"] = (
+            census_data["tot_rent_over35"] / census_data["B25070_001E"]
+    )
+    census_data["pct_with_disability"] = (
+            census_data["tot_with_disability"] / census_data["tot_pop"]
+    )
+    census_data["pct_eng_prof"] = (
+            census_data["tot_eng_prof"] / census_data["C16002_001E"]
+    )
+
+    return census_data
+
+
+def create_categories(row):
+    if row["indexed_vulnerability"] > 80:
+        return "Most Vulnerable"
+    if row["indexed_vulnerability"] > 60:
+        return "Medium-High Vulnerable"
+    if row["indexed_vulnerability"] > 40:
+        return "Medium Vulnerable"
+    if row["indexed_vulnerability"] > 20:
+        return "Medium-Low Vulnerable"
+    return "Least Vulnerable"
+
+
+def apply_weights(census_data):
+    # Applying the weights as defined in config.py
+
+    census_data["composite_vulnerability"] = 0
+    for item in WEIGHTS:
+        if "negative weight" in item:
+            census_data[item["column"]] = census_data[item["column"]] * -1
+        census_data[item["name"]] = (
+                ntile(census_data, item["column"], n=100) * item["weight"]
+        )
+        census_data["composite_vulnerability"] = (
+                census_data["composite_vulnerability"] + census_data[item["name"]]
+        )
+
+    # min-max normalization
+    census_data["indexed_vulnerability"] = (
+                                                   census_data["composite_vulnerability"]
+                                                   - census_data["composite_vulnerability"].min()
+                                           ) / (
+                                                   census_data["composite_vulnerability"].max()
+                                                   - census_data["composite_vulnerability"].min()
+                                           )
+    census_data["indexed_vulnerability"] = census_data["indexed_vulnerability"] * 100
+    census_data["eaz_type"] = census_data.apply(create_categories, axis=1)
+    return census_data
+
+
+def ntile(df, column, n):
```

Comment:
The [original work](https://github.com/cityofaustin/vulnerability_weighting_map/blob/main/app.R) used `ntile` in R's tidyverse. Pandas does not have an exact equivalent so this was necessary. This is actually an approximation, and it relies on the sort order of the dataframe so that also had to match the original work to get the 2019 index to line up.